fusedrug/eval/metrics/protein_sequences.py

```diff
@@ -152,3 +152,24 @@ def compare_strings(from_text: Any, to_text: Any, return_score: bool = False) ->
         return ans
 
     return counts
+
+
+def _naive_aar_no_indels(preds: List[str], target: List[str]) -> List[float]:
+    # evlauates the edit distance, normalized by the input length, between 2 strings of the same lengths (i.e., no insertions and deletion)
+    # incases where len(preds) != len(target) raises an exception
+    scores = []
+    for s1, s2 in zip(preds, target):
+
+        if len(s1) != len(s2):
+            raise Exception(
+                "Encountered different sequence lengths in prediction and target - can not eval naive AAR which assumes equal length"
+            )
+
+        score = 0.0
+        for c1, c2 in zip(s1, s2):
+            if c1 == c2:
+                score += 1
```

Comment:
Computing the score might be slow in-scale due string iteration. It's probably negligible :) 

If it's not, we can benchmark it with another implementations:
1. Using dedicated function from some lib.
2. Use numpy, something like (that's what I came up with):
```
>>> import numpy as np
>>> a = "apple"
>>> b = "assle"
>>> 
>>> a_arr = np.array(list(a))
>>> b_arr = np.array(list(b))
>>> 
>>> comp = np.char.equal(a_arr, b_arr)
>>> np.sum(~comp)
2
```

Reply:
I'm not sure my suggestion will be better in term of complexity tho... Just a thought (I'm in a verbose mode)

Reply:
I agree in general, `np.char.equal` specifically returns a single bool so not suitable. I guess other methods exist that we can use but might be an overkill and harder to read